### PR TITLE
Bug/add special fields for tokens redux

### LINKED ISSUE #2: Add special fields for tokens
When we add tokens, it would be nice to include some metadata related to them:

```
{ erc20: true, symbol: 'ETH', decimals: 18 }
```

This data could make this repo more easily consumable by [eth-token-tracker](https://github.com/flyswatter/eth-token-tracker).

### PR DESCRIPTION
**Pull request by flyswatter**
Mostly by @mintheredonethat

Opening on master branch for CI tests to run.

Re-submission of #3

Fixes #2

### CODE REVIEW

**Comment by flyswatter** [comment]
The last test failures seem to be that the address keys are not following the checksum format. An easy way to get these is to visit them on etherscan, and copy the mixed-case address from that page.

**Comment by flyswatter** [comment]
Also:

The `edg` and `trst` logos have transparent backgrounds. We might want to disallow that.

**Comment by flyswatter** [comment]
Or maybe we just recommend consumers add their own background defaults?

**Comment by mintheredonethat** [comment]
Sorry, I keep forgetting to test !

File: contract-map.json [hunk 1]
```diff
@@ -1,6 +1,181 @@
 {
+  "0xfa05A73FfE78ef8f1a739473e462c54bae6567D9": {
+    "name": "Lunyr Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified on Etherscan.

File: contract-map.json [hunk 2]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "lun.png",
+    "erc20": true,
+    "symbol": "LUN",
+    "decimals": 18
+  },
+  "0x5c543e7AE0A1104f78406C340E9C64FD9fCE5170": {
```
Comment (flyswatter) [inline contract-map.json]: Verified on Etherscan.
https://etherscan.io/address/0x5c543e7ae0a1104f78406c340e9c64fd9fce5170

File: contract-map.json [hunk 3]
```diff
@@ -1,6 +1,181 @@
...
+    "name": "vSlice",
+    "logo": "",
+    "erc20": true,
+    "symbol": "vSlice",
+    "decimals": 0
+  },
+  "0xFf3519eeeEA3e76F1F699CCcE5E23ee0bdDa41aC": {
+    "name": "Blockchain Capital",
```
Comment (flyswatter) [inline contract-map.json]: Verified on Etherscan
https://etherscan.io/address/0xff3519eeeea3e76f1f699ccce5e23ee0bdda41ac

File: contract-map.json [hunk 4]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "bcap.svg",
+    "erc20": true,
+    "symbol": "BCAP",
+    "decimals": 0
+  },
+  "0x6531f133e6DeeBe7F2dcE5A0441aA7ef330B4e53": {
+    "name": "Chronobank TIME",
```
Comment (flyswatter) [inline contract-map.json]: Verified on Etherscan
https://etherscan.io/address/0x6531f133e6deebe7f2dce5a0441aa7ef330b4e53

File: contract-map.json [hunk 5]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "chronobank.png",
+    "erc20": true,
+    "symbol": "TIME",
+    "decimals": 8
+  },
+  "0xE7775A6e9Bcf904eb39DA2b68c5efb4F9360e08C": {
+    "name": "Token-as-a-Service",
```
Comment (flyswatter) [inline contract-map.json]: Verified on Etherscan
0xE7775A6e9Bcf904eb39DA2b68c5efb4F9360e08C

File: contract-map.json [hunk 6]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "taas-ico.png",
+    "erc20": true,
+    "symbol": "TAAS",
+    "decimals": 6
+  },
+  "0xaAAf91D9b90dF800Df4F55c205fd6989c977E73a": {
+    "name": "Monolith TKN",
```
Comment (flyswatter) [inline contract-map.json]: Verified on Etherscan
https://etherscan.io/address/0xaAAf91D9b90dF800Df4F55c205fd6989c977E73a

File: contract-map.json [hunk 7]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "tkn.svg",
+    "erc20": true,
+    "symbol": "TKN",
+    "decimals": 8
+  },
+  "0x08711D3B02C8758F2FB3ab4e80228418a7F8e39c": {
+    "name": "Edgeless",
```
Comment (flyswatter) [inline contract-map.json]: Verified
0x08711D3B02C8758F2FB3ab4e80228418a7F8e39c

File: contract-map.json [hunk 8]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "",
+    "erc20": true,
+    "symbol": "EDG",
+    "decimals": 0
+  },
+  "0xf7B098298f7C69Fc14610bf71d5e02c60792894C": {
+    "name": "Guppy",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xf7b098298f7c69fc14610bf71d5e02c60792894c

File: contract-map.json [hunk 9]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "guppy.png",
+    "erc20": true,
+    "symbol": "GUP",
+    "decimals": 3
+  },
+  "0x4DF812F6064def1e5e029f1ca858777CC98D2D81": {
+    "name": "Xaurum",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x4df812f6064def1e5e029f1ca858777cc98d2d81

File: contract-map.json [hunk 10]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "xaurum_logo.svg",
+    "erc20": true,
+    "symbol": "XAUR",
+    "decimals": 8
+  },
+  "0xB9e7F8568e08d5659f5D29C4997173d84CdF2607": {
+    "name": "Swarm City Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xb9e7f8568e08d5659f5d29c4997173d84cdf2607

File: contract-map.json [hunk 11]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "swt.jpg",
+    "erc20": true,
+    "symbol": "SWT",
+    "decimals": 18
+  },
+  "0xCb94be6f13A1182E4A4B6140cb7bf2025d28e41B": {
+    "name": "Trustcoin",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xCb94be6f13A1182E4A4B6140cb7bf2025d28e41B

File: contract-map.json [hunk 12]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "",
+    "erc20": true,
+    "symbol": "TRST",
+    "decimals": 6
+  },
+  "0xBEB9eF514a379B997e0798FDcC901Ee474B6D9A1": {
+    "name": "Melon Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xbeb9ef514a379b997e0798fdcc901ee474b6d9a1

File: contract-map.json [hunk 13]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "melonport-1.svg",
+    "erc20": true,
+    "symbol": "MLN",
+    "decimals": 18
+  },
+  "0x960b236A07cf122663c4303350609A66A7B288C0": {
+    "name": "Aragon Network Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x960b236A07cf122663c4303350609A66A7B288C0

File: contract-map.json [hunk 14]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "aragon_isotype.svg",
+    "erc20": true,
+    "symbol": "ANT",
+    "decimals": 18
+  },
+  "0x667088b212ce3d06a1b553a7221E1fD19000d9aF": {
+    "name": "WINGS",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x667088b212ce3d06a1b553a7221E1fD19000d9aF

File: contract-map.json [hunk 15]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "wings_logo.svg",
+    "erc20": true,
+    "symbol": "WINGS",
+    "decimals": 18
+  },
+  "0xAf30D2a7E90d7DC361c8C4585e9BB7D2F6f15bc7": {
+    "name": "FirstBlood Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xaf30d2a7e90d7dc361c8c4585e9bb7d2f6f15bc7

File: contract-map.json [hunk 16]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "1st.svg",
+    "erc20": true,
+    "symbol": "1ST",
+    "decimals": 18
+  },
+  "0x607F4C5BB672230e8672085532f7e901544a7375": {
+    "name": "iEx.ec Network Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x607F4C5BB672230e8672085532f7e901544a7375

File: contract-map.json [hunk 17]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "rlc.png",
+    "erc20": true,
+    "symbol": "RLC",
+    "decimals": 9
+  },
+  "0xaeC2E87E0A235266D9C5ADc9DEb4b2E29b54D009": {
+    "name": "SingularDTV",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xaec2e87e0a235266d9c5adc9deb4b2e29b54d009

File: contract-map.json [hunk 18]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "singulardtv.svg",
+    "erc20": true,
+    "symbol": "SNGLS",
+    "decimals": 0
+  },
+  "0x888666CA69E0f178DED6D75b5726Cee99A87D698": {
+    "name": "ICONOMI",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x888666CA69E0f178DED6D75b5726Cee99A87D698

File: contract-map.json [hunk 19]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "iconomi.png",
+    "erc20": true,
+    "symbol": "ICN",
+    "decimals": 18
+  },
+  "0xE0B7927c4aF23765Cb51314A0E0521A9645F0E2A": {
+    "name": "Digix",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xE0B7927c4aF23765Cb51314A0E0521A9645F0E2A

File: contract-map.json [hunk 20]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "DGD.png",
+    "erc20": true,
+    "symbol": "DGD",
+    "decimals": 0
+  },
+  "0xC66eA802717bFb9833400264Dd12c2bCeAa34a6d": {
+    "name": "Maker",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xC66eA802717bFb9833400264Dd12c2bCeAa34a6d

File: contract-map.json [hunk 21]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "logo-maker-4.svg",
+    "erc20": true,
+    "symbol": "MKR",
+    "decimals": 0
+  },
+  "0x0D8775F648430679A709E98d2b0Cb6250d2887EF": {
+    "name": "Basic Attention Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x0D8775F648430679A709E98d2b0Cb6250d2887EF

File: contract-map.json [hunk 22]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "BAT_icon.svg",
+    "erc20": true,
+    "symbol": "BAT",
+    "decimals": 18
+  },
+  "0x6810e776880C02933D47DB1b9fc05908e5386b96": {
+    "name": "Gnosis Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x6810e776880C02933D47DB1b9fc05908e5386b96

File: contract-map.json [hunk 23]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "gnosis.svg",
+    "erc20": true,
+    "symbol": "GNO",
+    "decimals": 18
+  },
+  "0xD8912C10681D8B21Fd3742244f44658dBA12264E": {
+    "name": "Pluton",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xd8912c10681d8b21fd3742244f44658dba12264e

File: contract-map.json [hunk 24]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "plutus-god.svg",
+    "erc20": true,
+    "symbol": "PLU",
+    "decimals": 18
+  },
+  "0x48c80F1f4D53D5951e5D5438B54Cba84f29F32a5": {
+    "name": "Reputation",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0x48c80F1f4D53D5951e5D5438B54Cba84f29F32a5

File: contract-map.json [hunk 25]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "augur_logo.png",
+    "erc20": true,
+    "symbol": "REP",
+    "decimals": 18
+  },
+  "0xa74476443119A942dE498590Fe1f2454d7D4aC0d": {
+    "name": "Golem Network Token",
```
Comment (flyswatter) [inline contract-map.json]: Verified
https://etherscan.io/address/0xa74476443119a942de498590fe1f2454d7d4ac0d

File: contract-map.json [hunk 26]
```diff
@@ -1,6 +1,181 @@
...
+    "logo": "golem.svg",
+    "erc20": true,
+    "symbol": "GNT",
+    "decimals": 18
+  },
   "0x6090A6e47849629b7245Dfa1Ca21D94cd15878Ef": {
-    "logo": "ens.svg",
-    "name": "ENS Registrar"
+    "name": "ENS Registrar",
+    "logo": "ens.svg"
```
Comment (flyswatter) [inline contract-map.json]: The README says to not touch unmodified entries, but I'm letting it slide b/c it's a mega PR.

**Comment by flyswatter** [comment]
Published to npm as `eth-contract-metadata@1.1.0`.

**Pull request merged**